test/Microsoft.AspNet.Routing.Tests/RouteTest.cs

```diff
@@ -631,6 +631,60 @@ public void GetVirtualPath_Fail()
             Assert.Null(path);
         }
 
+        [Fact]
+        public void GetVirtualPath_ForList()
+        {
+            // Arrange
+            var route = CreateRoute("{controller}/{action}");
+            var context = CreateVirtualPathContext(
+                new { color = new List<string> { "red", "green", "blue" } },
+                new { controller = "Home", action = "Index" });
+
+            // Act
+            var pathData = route.GetVirtualPath(context);
+
+            // Assert
+            Assert.Equal(new PathString("/Home/Index?color=red&color=green&color=blue"), pathData.VirtualPath);
+            Assert.Same(route, pathData.Router);
+            Assert.Empty(pathData.DataTokens);
+        }
+
+        [Fact]
+        public void GetVirtualPath_ForList_Empty()
+        {
+            // Arrange
+            var route = CreateRoute("{controller}/{action}");
+            var context = CreateVirtualPathContext(
+                new { color = new List<string> { } },
+                new { controller = "Home", action = "Index" });
+
+            // Act
+            var pathData = route.GetVirtualPath(context);
+
+            // Assert
+            Assert.Equal(new PathString("/Home/Index"), pathData.VirtualPath);
+            Assert.Same(route, pathData.Router);
+            Assert.Empty(pathData.DataTokens);
+        }
+        
+        [Fact]
+        public void GetVirtualPath_ForList_StringWorkaround()
+        {
+            // Arrange
+            var route = CreateRoute("{controller}/{action}");
+            var context = CreateVirtualPathContext(
+                new { page = 1, color = new List<string> { "red", "green", "blue" }, message = "bum!" },
```

Comment:
Could you pick a text other than `bum!`? Our CI runs a tool on sources which looks for questionable content and this looks like something it would flag.